### PR TITLE
chore(template): adding regression info in the bug template

### PR DESCRIPTION
**Pull request by shikha372**
### Issue # (if applicable)

Closes .

### Reason for this change

Update bug template for users to inform about the code changes that resulted in breaking changes.

### Description of changes

- Adding checkbox to indicate whether this bug is a regression
- Add option of an input for adding last known CDK version
- Add GH workflow to run when an issue is opened or edit, workflow will add label `potential-regression` on the basis whether checkbox is selected in [issue template or not.](https://github.com/shikha372/aws-cdk/blob/regression_template/.github/ISSUE_TEMPLATE/bug-report.yml)

### Description of how you validated changes

Validated template [here](https://github.com/shikha372/aws-cdk/blob/regression_template/.github/ISSUE_TEMPLATE/bug-report.yml) 

Label : potential-regression (will be created after approval)
Label will removed if this option is unchecked and added if this option is checked.

Verified with sample issue in personal repo [here](https://github.com/shikha372/aws-cdk/actions/runs/10326894628), can be tested on sample [github issue ](https://github.com/shikha372/aws-cdk/issues/2)created in personal repo.
Result will be seen in [github actions](https://github.com/shikha372/aws-cdk/actions).

Sample issue: https://github.com/shikha372/aws-cdk/issues/2
Sample run: https://github.com/shikha372/aws-cdk/actions/runs/10326894628/job/28591182838

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -20,6 +20,21 @@ body:
...
+      options:
+        - label: Select this option if this issue appears to be a regression.
+          required: false
+  - type: input
+    id: working-version
+    attributes:
+      label: Last Known Working CDK Version
+      description: Specify the last known CDK version where this code was functioning as expected(if applicable).
+    validations:
+      required: false
```
Comment (GavinZZ) [inline .github/ISSUE_TEMPLATE/bug-report.yml:37]: For my learning purpose, how is `validations` different from 
```
options:
          required: false
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -20,6 +20,21 @@ body:
       description: What is the problem? A clear and concise description of the bug.
     validations:
       required: true
+  - type: checkboxes
+    id: regression
+    attributes:
+      label: Regression Issue
+      description:  What is a regression? If it worked in a previous version but doesn’t in the latest version, it’s considered a regression. In this case, Please provide specific version number in the report.
```
Comment (gracelu0) [inline .github/ISSUE_TEMPLATE/bug-report.yml:27]:
```suggestion
      description:  What is a regression? If it worked in a previous version but doesn’t in the latest version, it’s considered a regression. In this case, please provide specific version number in the report.
```

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 3]
```diff
@@ -20,6 +20,21 @@ body:
...
+      options:
+        - label: Select this option if this issue appears to be a regression.
+          required: false
+  - type: input
+    id: working-version
+    attributes:
+      label: Last Known Working CDK Version
+      description: Specify the last known CDK version where this code was functioning as expected(if applicable).
```
Comment (gracelu0) [inline .github/ISSUE_TEMPLATE/bug-report.yml:35]:
```suggestion
      description: Specify the last known CDK version where this code was functioning as expected (if applicable).
```

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 1 above)
Comment (shikha372) [reply .github/ISSUE_TEMPLATE/bug-report.yml:37]: haven't seen this field options in template but validation for the text area field is enforced when it is a necessary input required by the user.

Comment (GavinZZ) [reply .github/ISSUE_TEMPLATE/bug-report.yml:37]: Does that mean you're making the version field required?

Comment (shikha372) [reply .github/ISSUE_TEMPLATE/bug-report.yml:37]: making the field not required, options are seems to be for an array field from the [documentation](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#attributes-for-checkboxes)

Comment (GavinZZ) [reply .github/ISSUE_TEMPLATE/bug-report.yml:37]: I see, thanks for the clarification. Good to approve once addressing Grace's feedback.

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**